CMakeLists.txt

```diff
@@ -61,16 +61,16 @@ if(${CMAKE_SYSTEM_NAME} STREQUAL "OpenBSD")
   endif()
 endif()
 
-if(CMAKE_SYSTEM_NAME STREQUAL "SunOS")
+if(CMAKE_HOST_SYSTEM_NAME STREQUAL "SunOS")
```

Comment:
It's seems that the logic below only applies when the host and target systems are the same. CMake sets a [`CMAKE_CROSSCOMPILING`](https://cmake.org/cmake/help/latest/variable/CMAKE_CROSSCOMPILING.html) whenever the build has had `CMAKE_SYSTEM_NAME` manually set.

This seems like something that should be checked here:
```
if(CMAKE_HOST_SYSTEM_NAME STREQUAL "SunOS" AND NOT CMAKE_CROSSCOMPILING)
```